### PR TITLE
test: Refactor rpc_fundrawtransaction.py

### PR DESCRIPTION
**Pull request by akankshakashyap**
Made 2 changes to `tests/functional/rpc_fundrawtransaction.py`- 

1. **Remove test_simple_two_coins** - The current test_simple_two_coins and test_simple are identical and redundant. The test_simple_two_coins was changed in #6417  and asserts ensuring the use of two input coins were removed.
2. **Use changepos to get the change output** - In test_address_reuse the code to get the change output assumes that a coin with a value > 2 BTC was used to fund the transaction, and therefore change value will be greater than 1 BTC. This can fail if previous tests add a small coin. fundrawtransaction directly gives the index of change output as changepos, we use this to get the change output more reliably and cleanly.

### CODE REVIEW

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -103,9 +103,15 @@ def run_test(self):
         self.generate(self.nodes[2], 1)
         self.generate(self.nodes[0], 121)
 
+        # coins for simple tests
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+
+        self.generate(self.nodes[0], 1)
```
Comment (brunoerg) [inline test/functional/rpc_fundrawtransaction.py:111]: Did you duplicate this `self.generate(self.nodes[0], 1)`, any reason? You moved it to here but you also added it in L684.

Comment (akankshakashyap) [reply test/functional/rpc_fundrawtransaction.py:111]: Yes, generate mines a block so it has been duplicated to confirm the transactions preceding it.

Comment (brunoerg) [reply test/functional/rpc_fundrawtransaction.py:111]: It's only a suggestion, but I think this way would fit better. 
```diff
diff --git a/test/functional/rpc_fundrawtransaction.py b/test/functional/rpc_fundrawtransaction.py
index 338c31eea..71fda6362 100755
--- a/test/functional/rpc_fundrawtransaction.py
+++ b/test/functional/rpc_fundrawtransaction.py
@@ -103,13 +103,7 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.generate(self.nodes[2], 1)
         self.generate(self.nodes[0], 121)
 
-        # coins for simple tests
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
-
-        self.generate(self.nodes[0], 1)
-
+        self.test_watchonly()
         self.test_change_position()
         self.test_simple()
         self.test_simple_two_outputs()
@@ -132,7 +126,6 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.test_many_inputs_fee()
         self.test_many_inputs_send()
         self.test_op_return()
-        self.test_watchonly()
         self.test_all_watched_funds()
         self.test_option_feerate()
         self.test_address_reuse()
@@ -681,6 +674,11 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.nodes[0].lockunspent(False, [{"txid": self.watchonly_txid, "vout": self.watchonly_vout}])
 
         self.nodes[0].sendtoaddress(self.nodes[3].get_wallet_rpc(self.default_wallet_name).getnewaddress(), self.watchonly_amount / 10)
+                # coins for simple tests
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+
         self.generate(self.nodes[0], 1)
 
         inputs = []
```

**Comment by akankshakashyap** [comment]
> Looks like the test fails:

The failing test is testing wether fundrawtransaction selects enough fees to cover the transaction after signing. Changes made in this PR should at most change the available coins inside the wallet. 

I can't think of any reason for this test failing, other than there being some edge case in the fundrawtransaction's fees mechanism that these changes have chanced upon.

**Comment by akankshakashyap** [comment]
rebased.

**Comment by MarcoFalke** [comment]
I don't know either why it happens. Though, this can not be merged with a failing test.

**Comment by akankshakashyap** [comment]
The failing test passed after removing the 3rd commit.

**Comment by MarcoFalke** [comment]
You'll have to adjust the pull request description?

**Comment by achow101** [comment]
Closing this as it has not had a lot of conceptual support. If you are interested in continuing work on this, please leave a comment so that it can be reopened.

**Comment by MarcoFalke** [comment]
The changes look correct to me

**Pull request closed without merging**